packages/autorest.python/autorest/codegen/templates/serialization.py.jinja2

```diff
@@ -732,6 +732,8 @@ class Serializer(object):
 
             if kwargs.get("skip_quote") is True:
                 output = str(output)
+                # https://github.com/Azure/autorest.python/issues/2063
+                output = output.replace("{", quote("{")).replace("}", quote("}"))
```

Comment:
with reference to https://github.com/Azure/azure-sdk-for-python/pull/31375/files#diff-179c73d96911cfd502138f0f4b39197df61a64a0c1e5af26225ba9bd82e3a4a6